hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SNDArray.scala

```diff
@@ -626,6 +627,63 @@ object SNDArray {
       work.firstDataAddress, lwork.toI))
     cb.ifx(info.cne(0), cb._fatal(s"LAPACK error DGEQRF. Error code = ", info.toS))
   }
+
+  def syevr_query(cb: EmitCodeBuilder, jobz: String, uplo: String, n: Value[Int], region: Value[Region]): (SizeValue, SizeValue) = {
+    val WorkAddress = cb.memoize(region.allocate(8L, 8L))
```

Comment:
for my information, what's with all the memoization?

Reply:
That's a fundamental part of our codegen abstraction. Would be happy to discuss in person. Short version is: `region.alllocate` returns a `Code`, which is a wrapper around some bytecode, and `cb.memoize` adds the bytecode to the `CodeBuilder` and returns a `Value`, which is just a reference to data on the stack.

Reply:
Thanks!